data/installed-tests/at90usbkey124.metainfo.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2017 Richard Hughes <richard@hughsie.com> -->
+<component type="firmware">
+  <id>com.acme.at90usbkey.firmware</id>
+  <name>AT90USBKEY Firmware</name>
+  <summary>Firmware for the ACME AT90USBKEY</summary>
+  <description>
+    <p>
+      Updating the firmware on your AT90USBKEY improves performance and
+      adds new features.
+    </p>
+  </description>
+  <provides>
+    <!-- USB\VID_273F&PID_2000 -->
+    <firmware type="flashed">c1874c52-5f6a-5864-926d-ea84bcdc82ea</firmware>
+  </provides>
+  <url type="homepage">http://www.acme.com/</url>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0+</project_license>
```

Comment:
Wait, the binary included here is GPL?  Could the source be distributed and binary built then?

Reply:
Well, I *think* it's GPL. I've used the "Open Source" LUFA mouse demo code then made substantial additions to add a DFU runtime interface. The actual code is https://github.com/hughsie/fwupd-test-firmware/blob/master/AVR/Mouse.c -- is that MIT? I'm unsure what the license of the binary then becomes.

Although you can generate this yourself, the SHA1 is going to be different (built times etc) as the avr-gcc compiler isn't as good at the vanilla gcc compiler at the reproducible builds thing. In the future fwupd-test-firmware is going to also need the nonfree AVR32 compiler, and the nonfree Microchip xc8 compiler, so I didn't want to complicate the build of fwupd with all that.

Reply:
Ah I see.  How about this to sidestep the possible licensing mess:
1) Move the "precompiled' .hex files to that repo
2) For those two specifically, curl/wget the binaries in fwupdmgr.sh installed-tests script
3) generate the cab file in the fwupdmgr.sh installed-tests script

That should make it easier to keep all of fwupd GPL and not have to worry about compiler and licensing limitations for tests.

Reply:
Note, we don't actually need the binary files in fwupd at all -- the .cab files get downloaded from the LVFS for the hardware tests. I just included them so it would be easy to regenerate the .cab files for the LVFS, in case I ever get hit by a bus or something. If you think it's best I can move those and the other DFU test firmware there as well.

Reply:
Oh, then yes I think that's a good idea to do.

Reply:
Done in master.

Reply:
great, thanks!